webui/react/src/hooks/useAuthCheck.ts

```diff
@@ -42,18 +43,24 @@ const useAuthCheck = (): (() => Promise<boolean>) => {
     const authToken = jwtToken ?? cookieToken ?? globalStorage.authToken;
 
     /*
-     * If auth token found, update the API bearer token and validate it with the current user API.
+     * If an auth token is found, validate it with the current user API, then update
+     * the API bearer token and set the user as authenticated.
      * If an external login URL is provided, redirect there.
      * Otherwise mark that we checked the auth and skip auth token validation.
      */
 
     if (authToken) {
-      updateBearerToken(authToken);
-
-      Observable.batch(() => {
-        authStore.setAuth({ isAuthenticated: true, token: authToken });
-        authStore.setAuthChecked();
-      });
+      try {
+        await getCurrentUser({});
+        updateBearerToken(authToken);
+        Observable.batch(() => {
+          authStore.setAuth({ isAuthenticated: true, token: authToken });
+        });
```

Comment:
**question:** is there a need for `Observable.batch` if there's only one observable being updated inside?